### ISSUE TITLE
feat: A Dart Frog Android Studio/ IntelliJ Extension 

### ISSUE THREAD

**Issue by kaljitism** [VeryGoodOpenSource/dart_frog#1326]
**Description**

I have found that there is a extension of Dart Frog in [Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=VeryGoodVentures.dart-frog). I am a user of Android Studio and couldn't find any such plugin on Jetbrains Marketplace. I wonder if the team is working on an Android Studio Plugin?

**Comment by alestiago** [comment]
Hi @kaljitism ! Thanks for opening an issue. There is no official Android Studio/IntelliJ extension. You might want to look into this related issues #286 and #775, I hope this helps 🙌

**Comment by tomarra** [comment]
Hi @kaljitism 👋 thanks for opening this issue. As pointed out by @alestiago we have a duplicate ticket on this already which we have closed out as it's not a priority for us. This is due to that original ticket not having many votes or activity and that at VGV we don't typically use Android Studio in our workflow. 

While it would be great to have a contribution made by the community for this it would then also keep us on the hook for maintenance and support for that extension as well if/when things break. If this is something that you are willing to sign up for in the long term (meaning things like tickets that come up get assigned to you to resolve) then you should feel free to work on this. I think the best way to do so would be to work on a proposal on how you would like to implement it, document it in this ticket, and then ask for feedback before starting to write code. In the past we have had to do a lot of back and forth in PR's when a discussion before implementation starts would have saved everyone a lot of time.

**Comment by tomarra** [comment]
Also reworking the labels here. Removing `duplicate` given that the other ticket is closed, adding `feature` given that its a new request, adding `p3` as it's not on our priority list.

**Comment by kaljitism** [comment]
> While it would be great to have a contribution made by the community for this it would then also keep us on the hook for maintenance and support for that extension as well if/when things break. If this is something that you are willing to sign up for in the long term (meaning things like tickets that come up get assigned to you to resolve) then you should feel free to work on this. I think the best way to do so would be to work on a proposal on how you would like to implement it, document it in this ticket, and then ask for feedback before starting to write code. In the past we have had to do a lot of back and forth in PR's when a discussion before implementation starts would have saved everyone a lot of time.

Thanks for replying @tomarr

**Comment by kaljitism** [comment]
> While it would be great to have a contribution made by the community for this it would then also keep us on the hook for maintenance and support for that extension as well if/when things break. If this is something that you are willing to sign up for in the long term (meaning things like tickets that come up get assigned to you to resolve) then you should feel free to work on this. I think the best way to do so would be to work on a proposal on how you would like to implement it, document it in this ticket, and then ask for feedback before starting to write code. In the past we have had to do a lot of back and forth in PR's when a discussion before implementation starts would have saved everyone a lot of time.

Thanks @tomarra  for replying! If I develop the Android Studio plugin, I'll definitely sign up for all the taking care of all the Issues that arises and also take the responsibility to regularly maintain it and push update as the Frog framework is updated. I did a little bit of research and worked with the VSCode plugin to be familiar with what is happening. 

I am learning how Android Studio plugins are implemented and it uses Kotlin, Gradle, XML and [Plugin SDK](https://plugins.jetbrains.com/docs/intellij/welcome.html) for IntelliJ Platform. Once, I am done with all these and confident enough to implement the same functionality as VSCode/ TypeScript plugin, I will create a plan of implementation and update it here!
